core/shared/src/main/scala/interface.scala

```diff
@@ -155,6 +155,26 @@ abstract class CaseClass[Typeclass[_], Type] private[magnolia] (
   final def construct[Return](makeParam: Param[Typeclass, Type] => Return): Type =
     rawConstruct(parameters map makeParam)
 
+  /**
+   * Like construct but allows each parameter to fail with an error.
+   *
+   * @see construct
+   */
+  final def econstruct[E <: AnyRef, Return](makeParam: Param[Typeclass, Type] => Either[E, Return]): Either[E, Type] = {
+    // poor man's scalaz.Traverse
+    try {
+      val bits = parameters.map { p =>
```

Comment:
`bits` is misleading, I think. How about writing it as a point-free expression as a way to avoid coming up with a unique name? ;)

Reply:
I also don't think the name `econstruct` is clear enough. The `e` could stand for "either" or "exception". Let's go with `eitherConstruct`, unless you can suggest anything better...